db_test.go

```diff
@@ -85,41 +85,65 @@ func TestQuery(t *testing.T) {
 		},
 	}
 	expectedTypeNames := []string{"varchar", "smallint", "integer", "bigint", "boolean", "float", "double", "varchar", "timestamp", "date", "decimal"}
+	expectedTypeNameGzipDLs := []string{"string", "smallint", "int", "bigint", "boolean", "float", "double", "string", "timestamp", "date", "decimal(11,5)"}
 	harness.uploadData(expected)
 
-	rows := harness.mustQuery("select * from %s", harness.table)
-	index := -1
-	for rows.Next() {
-		index++
-
-		var row dummyRow
-		require.NoError(t, rows.Scan(
-			&row.NullValue,
-
-			&row.SmallintType,
-			&row.IntType,
-			&row.BigintType,
-			&row.BooleanType,
-			&row.FloatType,
-			&row.DoubleType,
-			&row.StringType,
-			&row.TimestampType,
-			&row.DateType,
-			&row.DecimalType,
-		))
-
-		assert.Equal(t, expected[index], row, fmt.Sprintf("index: %d", index))
-
-		types, err := rows.ColumnTypes()
-		assert.NoError(t, err, fmt.Sprintf("index: %d", index))
-		for i, colType := range types {
-			typeName := colType.DatabaseTypeName()
-			assert.Equal(t, expectedTypeNames[i], typeName, fmt.Sprintf("index: %d", index))
-		}
+	resultModes := []ResultMode {
+		ResultModeAPI,
+		ResultModeDL,
+		ResultModeGzipDL,
 	}
 
-	require.NoError(t, rows.Err(), "rows.Err()")
-	require.Equal(t, 3, index+1, "row count")
+	for _, resultMode := range resultModes {
+		ctx := context.Background()
+		switch resultMode {
+		case ResultModeAPI:
+			ctx = SetAPIMode(ctx)
+		case ResultModeDL:
+			ctx = SetDLMode(ctx)
+		case ResultModeGzipDL:
+			ctx = SetGzipDLMode(ctx)
+		}
+
+		rows := harness.mustQuery(ctx, "select * from %s", harness.table)
+		index := -1
+		for rows.Next() {
+			index++
+
+			var row dummyRow
+			require.NoError(t, rows.Scan(
+				&row.NullValue,
+
+				&row.SmallintType,
+				&row.IntType,
+				&row.BigintType,
+				&row.BooleanType,
+				&row.FloatType,
+				&row.DoubleType,
+				&row.StringType,
+				&row.TimestampType,
+				&row.DateType,
+				&row.DecimalType,
+			))
+
+			assert.Equal(t, expected[index], row, fmt.Sprintf("resultMode:%v, index: %d", resultMode, index))
```

Comment:
[nits] I'm a bit concerned about `resultMode:%v` doesn't have space, and `index: %d` has space. Is it better to make it consistent? or Is there any criteria about whether to add space or not?

Reply:
I fixed it.